common/rpc/params.go

```diff
@@ -83,8 +84,8 @@ func NewParams(serviceName string, config *config.Config) (Params, error) {
 
 	return Params{
 		ServiceName:       serviceName,
-		TChannelAddress:   fmt.Sprintf("%v:%v", listenIP, serviceConfig.RPC.Port),
-		GRPCAddress:       fmt.Sprintf("%v:%v", listenIP, serviceConfig.RPC.GRPCPort),
+		TChannelAddress:   net.JoinHostPort(listenIP.String(), strconv.Itoa(serviceConfig.RPC.Port)),
+		GRPCAddress:       net.JoinHostPort(listenIP.String(), strconv.Itoa(serviceConfig.RPC.GRPCPort)),
```

Comment:
Just curious, is there anything different functionally? or just a more standard way to do.

Reply:
IpV6 addresses must be represented with the port in this way [ipaddr]:port. This function does the job by ensuring that square bracket are always there if an IPv6 is provided in input.

Reply:
And it is of course more standard as it support both v4 and v6. Just concatenating ip:port does not work in v6 ::1:3306 is invalid as example.